src/components/OrderBookWidget.tsx

```diff
@@ -157,18 +162,19 @@ const draw = (baseToken: TokenDetails, quoteToken: TokenDetails, dataSource: str
 }
 
 const OrderBookWidget: React.FC<OrderBookProps> = props => {
-  const { baseToken, quoteToken } = props
-  const { networkId } = useWalletConnection()
-  const [chart, setChart] = useState<am4charts.XYChart | null>(null)
+  const { baseToken, quoteToken, networkId } = props
+  const mountPoint = useRef<HTMLDivElement>(null)
+
   useEffect(() => {
-    if (chart) {
-      chart.dispose()
-    }
-    setChart(draw(baseToken, quoteToken, orderbookUrl(baseToken, quoteToken, networkId)))
+    if (!mountPoint.current) return
+
+    const chart = draw(mountPoint.current, baseToken, quoteToken, orderbookUrl(baseToken, quoteToken, networkId))
+
+    return (): void => chart.dispose()
```

Comment:
I think we still need to dispose on every redraw (e.g. when tokens change)

Reply:
And that's exactly how  `useEffect` works
```ts
useEffect(() => {
...
    return (): void => chart.dispose() //runs on clean-up stage
  }, [baseToken, quoteToken, networkId]) //deps
```
deps change -> cleanup function runs -> actual function runs -> deps change -> clean-up ...